terraform/main.tf

```diff
@@ -31,4 +31,95 @@ resource "azurerm_storage_account" "front_end_storage_account" {
   static_website {
     index_document = "index.html"
   }
+}
+
+resource "azurerm_resource_group" "product_service_rg" {
+  location = "northeurope"
+  name     = "rg-product-service-sand-ne-001"
+}
+
+resource "azurerm_storage_account" "products_service_fa" {
+  name     = "garegin789898u78"
+  location = "northeurope"
+
+  account_replication_type = "LRS"
+  account_tier             = "Standard"
+  account_kind             = "StorageV2"
+
+  resource_group_name = azurerm_resource_group.product_service_rg.name
+}
+
+resource "azurerm_storage_share" "products_service_fa" {
+  name  = "fa-products-service-share"
+  quota = 2
+
+  storage_account_name = azurerm_storage_account.products_service_fa.name
+}
+
+resource "azurerm_service_plan" "product_service_plan" {
+  name     = "asp-product-service-sand-ne-001"
+  location = "northeurope"
+
+  os_type  = "Windows"
+  sku_name = "Y1"
+
+  resource_group_name = azurerm_resource_group.product_service_rg.name
+}
+
+resource "azurerm_application_insights" "products_service_fa" {
+  name             = "appins-fa-products-service-sand-ne-001"
+  application_type = "web"
+  location         = "northeurope"
+
+
+  resource_group_name = azurerm_resource_group.product_service_rg.name
+}
+
+resource "azurerm_windows_function_app" "products_service" {
+  name     = "fa-products-service-ne-001aaaaaa"
+  location = "northeurope"
+
+  service_plan_id     = azurerm_service_plan.product_service_plan.id
+  resource_group_name = azurerm_resource_group.product_service_rg.name
+
+  storage_account_name       = azurerm_storage_account.products_service_fa.name
+  storage_account_access_key = azurerm_storage_account.products_service_fa.primary_access_key
+
+  functions_extension_version = "~4"
+  builtin_logging_enabled     = false
+
+  site_config {
+    always_on = false
+
+    application_insights_key               = azurerm_application_insights.products_service_fa.instrumentation_key
+    application_insights_connection_string = azurerm_application_insights.products_service_fa.connection_string
+
+    # For production systems set this to false
+    use_32_bit_worker = false
```

Comment:
when using an App Service Plan in the Free or Shared Tiers use_32_bit_worker_process must be set to true.

Reply:
@AliaksandrKhatsko please check my change